node/src/protocol/mod.rs

```diff
@@ -86,9 +88,13 @@ impl ConsensusCtx for &Ctx {
     fn sign_sk(&self) -> &near_crypto::SecretKey {
         &self.sign_sk
     }
+
+    fn secret_storage(&self) -> &SecretNodeStorageBox {
+        &self.secret_storage
+    }
 }
 
-impl CryptographicCtx for &Ctx {
+impl CryptographicCtx for &mut Ctx {
```

Comment:
nothing to change here, but now that I think about it, why weren't we just passing around a concrete `Ctx` object instead of requiring specific `*Ctx` interfaces? Doesn't seem like we're getting much out of having traits for this purpose but maybe I'm missing something

Reply:
The initial idea was to be able to "mock" specific methods that are necessary for that component without having to instantiate other stuff. Not sure if that makes sense anymore as most of the methods are used everywhere

Reply:
yeah, I kept having to add new interface members to each of the traits, so probably need to clean this up later. made a ticket: https://github.com/near/mpc-recovery/issues/406